client/app/telecom/telephony/alias/special/rsva/telecom-telephony-alias-special-rsva.controller.js

```diff
@@ -1,32 +1,92 @@
+    this.$state = $state;
+    this.$stateParams = $stateParams;
     this.$translate = $translate;
     this.costs = costs;
     this.OvhApiOrder = OvhApiOrder;
-    this.OvhApiTelephony = OvhApiTelephony;
-    this.OvhApiTelephonyService = OvhApiTelephonyService;
     this.tucTelephonyBulk = tucTelephonyBulk;
     this.TucToast = TucToast;
-    this.URLS = URLS;
+    this.tucVoipService = tucVoipService;
+    this.tucVoipServiceAlias = tucVoipServiceAlias;
   }
 
   $onInit() {
-    this.rsva = {};
-    this.rsvaForm = {};
+    this.serviceInfos = {
+      billingAccount: this.$stateParams.billingAccount,
+      serviceName: this.$stateParams.serviceName,
+    };
+
+    this.tariffBearingPrice = `${this.costs.rsva.tariffBearing.value} ${(this.costs.rsva.tariffBearing.currencyCode)}`;
+    this.initializeBulkData();
+
+    this.loading = true;
+    return this.$q.all({
+      directory: this.tucVoipService.getServiceDirectory(this.serviceInfos),
+      rateCodeInfos: this.getRateCodeInformations(this.serviceInfos),
+      rsvaInfos: this.tucVoipServiceAlias.getRSVAInformations(this.serviceInfos),
+      schema: this.OvhApiOrder.v6().schema().$promise,
+    })
+      .then(({
+        directory, rateCodeInfos, rsvaInfos, schema,
+      }) => {
+        this.rateCodeInfos = rateCodeInfos.rateCodeInfos;
+        this.directory = directory;
+
+        this.allowedRateCodes = rateCodeInfos.allowedRateCodes;
+        this.currentRateCode = rateCodeInfos.currentRateCode;
 
-    this.links = _.pick(this.URLS, ['transferTable', 'deontology', 'graphicCharter', 'svaToSvaPlus']);
-    this.tariffBearingPrice = `${this.costs.rsva.tariffBearing.value} ${(this.costs.rsva.tariffBearing.currencyCode === 'EUR' ? '€' : this.costs.rsva.tariffBearing.currencyCode)}`;
+        this.rateCode = this.allowedRateCodes
+          .find(({ code }) => code === this.currentRateCode.rateCode);
+        this.pricingType = this.rateCode.pricePerMinuteWithoutTax.value > 0 ? 'minute' : 'call';
+        this.updateRateCodesByType();
 
-    this.isLoading = true;
+        this.scheduledRateCode = rateCodeInfos.scheduledRateCode;
 
-    // Initialize bulk datas
+        if (this.scheduledRateCode) {
+          this.tariffBearingPrice = this.scheduledRateCode.updateRateCodePriceWithoutTax.value
+            ? this.scheduledRateCode.updateRateCodePriceWithoutTax.text : this.tariffBearingPrice;
+        }
+
+        if (_.has(schema, 'models[\'telephony.NumberSpecialTypologyEnum\'].enum')) {
+          this.regExp = new RegExp(`^${directory.country}_`);
+          this.typologies = schema.models['telephony.NumberSpecialTypologyEnum'].enum
+            .filter(element => element.match(this.regExp))
+            .map(typology => ({
+              value: typology,
+              label: `${this.$translate.instant(`telephony_alias_special_rsva_infos_typology_${typology.replace(this.regExp, '')}_label`)}`,
+            }));
+
+          this.typology = this.typologies
```

Comment:
I know i'm a huge pain in the bottom because it looks like I always change my mind, but in this case, Lodash might be better, what do you think ? 😅 

```suggestion
          this.typology = _.find(this.typologies, { value: `${directory.country}_${rsvaInfos.typology.replace(directory.country, '')}`})
```

Reply:
As we discuss, no problem, lodash is in this case easier to read :)